src/sqlancer/stonedb/oracle/StoneDBFuzzOracle.java

```diff
@@ -0,0 +1,82 @@
+package sqlancer.stonedb.oracle;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sqlancer.Randomly;
+import sqlancer.common.ast.newast.ColumnReferenceNode;
+import sqlancer.common.ast.newast.NewBinaryOperatorNode;
+import sqlancer.common.ast.newast.NewPostfixTextNode;
+import sqlancer.common.ast.newast.Node;
+import sqlancer.common.ast.newast.TableReferenceNode;
+import sqlancer.common.oracle.TestOracle;
+import sqlancer.common.query.ExpectedErrors;
+import sqlancer.common.query.SQLQueryAdapter;
+import sqlancer.stonedb.StoneDBErrors;
+import sqlancer.stonedb.StoneDBProvider.StoneDBGlobalState;
+import sqlancer.stonedb.StoneDBSchema;
+import sqlancer.stonedb.StoneDBSchema.StoneDBColumn;
+import sqlancer.stonedb.StoneDBSchema.StoneDBDataType;
+import sqlancer.stonedb.StoneDBSchema.StoneDBTable;
+import sqlancer.stonedb.StoneDBSchema.StoneDBTables;
+import sqlancer.stonedb.StoneDBToStringVisitor;
+import sqlancer.stonedb.ast.StoneDBExpression;
+import sqlancer.stonedb.ast.StoneDBJoin;
+import sqlancer.stonedb.ast.StoneDBSelect;
+import sqlancer.stonedb.gen.StoneDBExpressionGenerator;
+import sqlancer.stonedb.gen.StoneDBExpressionGenerator.StoneDBBinaryLogicalOperator;
+import sqlancer.stonedb.gen.StoneDBExpressionGenerator.StoneDBCastOperation;
+
+public class StoneDBFuzzOracle implements TestOracle<StoneDBGlobalState> {
+    private final StoneDBGlobalState globalState;
+    private final StoneDBSchema schema;
+    private final ExpectedErrors errors = new ExpectedErrors();
+
+    public StoneDBFuzzOracle(StoneDBGlobalState globalState) {
+        this.globalState = globalState;
+        this.schema = globalState.getSchema();
+        StoneDBErrors.addExpectedExpressionErrors(errors);
+    }
+
+    @Override
+    public void check() throws Exception {
+        StoneDBTables randomTables = schema.getRandomTableNonEmptyTables();
+        List<StoneDBColumn> columns = randomTables.getColumns();
+        StoneDBExpressionGenerator gen = new StoneDBExpressionGenerator(globalState).setColumns(columns);
+        Node<StoneDBExpression> randomWhereCondition = gen.generateExpression();
+        List<StoneDBTable> tables = randomTables.getTables();
+        List<TableReferenceNode<StoneDBExpression, StoneDBTable>> tableList = tables.stream()
+                .map(t -> new TableReferenceNode<StoneDBExpression, StoneDBTable>(t)).collect(Collectors.toList());
+        List<Node<StoneDBExpression>> joins = StoneDBJoin.getJoins(tableList, globalState);
+
+        StoneDBSelect select = new StoneDBSelect();
+        Node<StoneDBExpression> asText = new NewPostfixTextNode<>(
+                new StoneDBCastOperation(
+                        new NewBinaryOperatorNode<>(new NewPostfixTextNode<>(randomWhereCondition, " IS NOT NULL "),
```

Comment:
Why do we need this specific expression in the fetch clause?

Reply:
no special reasons. I just copied from the NoRec implementation. 😂

If it is not necessary, we can change it it `*` ?

Reply:
Yes, that, and perhaps selecting some random column subset with another probability.

Reply:
that make sense. have done.